### ISSUE TITLE
Swagger plugin v4 enum as default value

### ISSUE THREAD

**Issue by tsm91** [nestjs/swagger#542]
## I'm submitting a...

<pre><code>
[ ] Regression 
[x] Bug report
[ ] Feature request
[ ] Documentation issue or request
[ ] Support request => Please do not submit support request here, instead post your question on Stack Overflow.
</code></pre>

## Current behavior

I want to set a default enum value for a dto's key. If the enum is being 
imported from another file, then swagger plugin throws and error. If the enum 
is defined in the same file, then everything works.

## Minimal reproduction of the problem with instructions

this fails: 

```ts
import { IsEnum, IsOptional } from 'class-validator';
import { MeasurementSystem } from '../../core/enums/measurement-system.enum';

export class SomeDto {
  @IsEnum(MeasurementSystem)
  @IsOptional()
  readonly measurementSystem?: MeasurementSystem = MeasurementSystem.Imperial;
}
```

this works:
```ts
import { IsEnum, IsOptional } from 'class-validator';
// import { MeasurementSystem } from '../../core/enums/measurement-system.enum';

enum MeasurementSystem { Metric = 'metric', Imperial = 'imperial'}
export class SomeDto {
  @IsEnum(MeasurementSystem)
  @IsOptional()
  readonly measurementSystem?: MeasurementSystem = MeasurementSystem.Imperial;
}
```

## Error i get in the console during compilation time
```bash
(node:62539) UnhandledPromiseRejectionWarning: ReferenceError: MeasurementSystem is not defined
    at Function._OPENAPI_METADATA_FACTORY (/Users/aegyed/dev2/projects/hp-api/dist/user/dto/create-user-request.dto.js:22:468)
    at ModelPropertiesAccessor.applyMetadataFactory (/Users/aegyed/dev2/projects/hp-api/node_modules/@nestjs/swagger/dist/services/model-properties-accessor.js:26:93)
    at SchemaObjectFactory.exploreModelSchema (/Users/aegyed/dev2/projects/hp-api/node_modules/@nestjs/swagger/dist/services/schema-object-factory.js:54:38)
    at /Users/aegyed/dev2/projects/hp-api/node_modules/@nestjs/swagger/dist/services/schema-object-factory.js:29:36
    at Array.map (<anonymous>)
    at SchemaObjectFactory.createFromModel (/Users/aegyed/dev2/projects/hp-api/node_modules/@nestjs/swagger/dist/services/schema-object-factory.js:19:27)
    at exports.exploreApiParametersMetadata (/Users/aegyed/dev2/projects/hp-api/node_modules/@nestjs/swagger/dist/explorers/api-parameters.explorer.js:32:55)
    at /Users/aegyed/dev2/projects/hp-api/node_modules/@nestjs/swagger/dist/swagger-explorer.js:67:45
    at Array.reduce (<anonymous>)
    at /Users/aegyed/dev2/projects/hp-api/node_modules/@nestjs/swagger/dist/swagger-explorer.js:66:99
    at /Users/aegyed/dev2/projects/hp-api/node_modules/lodash/lodash.js:13401:38
    at /Users/aegyed/dev2/projects/hp-api/node_modules/lodash/lodash.js:4905:15
    at baseForOwn (/Users/aegyed/dev2/projects/hp-api/node_modules/lodash/lodash.js:2990:24)
    at Function.mapValues (/Users/aegyed/dev2/projects/hp-api/node_modules/lodash/lodash.js:13400:7)
    at MapIterator.iteratee (/Users/aegyed/dev2/projects/hp-api/node_modules/@nestjs/swagger/dist/swagger-explorer.js:66:45)
    at MapIterator.next (/Users/aegyed/dev2/projects/hp-api/node_modules/iterare/lib/map.js:13:39)
(node:62539) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 1)
(node:62539) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```

## Environment

<pre><code>
Nest version: 6.14.1

 
For Tooling issues:
- Node version: v12.14.1
- @nestjs/swagger version: v4.2.3
- Platform:  Mac

**Comment by IGassmann** [comment]
Probably a duplicate of https://github.com/nestjs/swagger/issues/466.

**Comment by kamilmysliwiec** [comment]
It should be fixed in 4.2.4 :)
